pygmt/accessors.py

```diff
@@ -29,12 +31,16 @@ def __init__(self, xarray_obj):
         self._obj = xarray_obj
         try:
             self._source = self._obj.encoding["source"]  # filepath to NetCDF source
+            if not Path(self._source).exists():
+                raise FileNotFoundError(
+                    f"Grid source file {self._source} doesn't exist."
+                )
             # Get grid registration and grid type from the last two columns of
             # the shortened summary information of `grdinfo`.
             self._registration, self._gtype = map(
                 int, grdinfo(self._source, per_column="n").split()[-2:]
             )
-        except (KeyError, ValueError):
+        except (KeyError, ValueError, FileNotFoundError):
             self._registration = 0  # Default to Gridline registration
             self._gtype = 0  # Default to Cartesian grid type
```

Comment:
I wonder if we should change these lines to the following codes, which I think are more readable:
```
        self._source = self._obj.encoding.get("source")
        if self._source is not None and Path(self._source).exists():
            try:
                # Get grid registration and grid type from the last two columns of
                # the shortened summary information of `grdinfo`.
                self._registration, self._gtype = map(
                    int, grdinfo(self._source, per_column="n").split()[-2:]
                )
            except ValueError:
                self._registration = 0
                self._gtype = 0
        else:
            self._registration = 0
            self._gtype = 0
```

Reply:
So we will silently fallback to registration=0 and gtype=0 if `grdinfo` cannot determine the grid registration and type? Ok I guess, but please add the code comments back :slightly_smiling_face: